rts/motoko-rts/src/types.rs

```diff
@@ -452,6 +453,22 @@ impl Array {
     }
 }
 
+#[rustfmt::skip]
+#[repr(C)] // See the note at the beginning of this module
+pub struct Region {
+    pub header: Obj,
+    pub id: u16,
+    pub page_count: u64,
+    pub vec_ptr: *mut Blob, // Blob knows its own capacity
```

Comment:
```suggestion
    pub vec_ptr: *mut Blob, // Blob knows its own capacity
```
This field should not be hear, but stored at payload_addr, I think - it's not part of the header but in the sole payload field.